torch/lib/THD/base/data_channels/DataChannelNccl.hpp

```diff
@@ -0,0 +1,173 @@
+#pragma once
+
+#include "../DataChannel.hpp"
+#include "DataChannelUtils.hpp"
+
+#include <nccl.h>
+
+#include <utility>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+
+#define NCCL_CHECK(cmd) do {                                  \
+  ncclResult_t error = cmd;                                   \
+  if (error != ncclSuccess) {                                 \
+    std::string err = "NCCL error in: " +                     \
+                      std::string(__FILE__) + ":" +           \
+                      std::to_string(__LINE__) + ", " +       \
+                      std::string(ncclGetErrorString(error)); \
+    throw std::runtime_error(err);                            \
+  }                                                           \
+} while (0)
+
+
+namespace thd {
+
+struct DataChannelNccl : DataChannel {
+  struct RequestNccl : DataChannel::Request {
+
+    RequestNccl(QueueWorker::Request&& request);
+    virtual ~RequestNccl();
+
+    virtual bool isCompleted() override;
+    virtual void wait() override;
+
+  private:
+    QueueWorker::Request _request;
+  };
```

Comment:
I don't think you need to implement anything (no memebers are necessary) for requests if you don't support `send` and `recv` anyway